### PR TITLE
fix: the Abs, Ceil and Floor methods now return proper types

### LINKED ISSUE #3790: Breaking change: fix return types on inbuilt functions
We know we have some UDFs where the return type is wrong, e.g. `floor(integer)` returns a `DOUBLE`, where as it should return an integer type.  There are likely others.  We should fix these in the next major version.

### PR DESCRIPTION
**Pull request by agavra**
BREAKING CHANGE: abs, ceil and floor will now return types aligned with
other databases systems (i.e. the same type as the input). Previously
these udfs would always return Double.

fixes #3790 

### Testing done 

- Updated `math.json`

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Confluent Inc.
```
Comment (stevenpyzhang) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java]: nit: 2019

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.KsqlException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+
+@UdfDescription(name = "Ceil", description = Ceil.DESCRIPTION)
+public class Ceil {
+
+  static final String DESCRIPTION = "Returns the largest integer greater than or equal to the "
```
Comment (stevenpyzhang) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java]: Shouldn't this say `smallest integer`

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    return val == null
+        ? null
+        : val.setScale(0, RoundingMode.CEILING).setScale(val.scale(), RoundingMode.UNNECESSARY);
+  }
+
+  @UdfSchemaProvider
+  public SqlType ceilDecimalProvider(final List<SqlType> params) {
+    final SqlType s = params.get(0);
+    if (s.baseType() != SqlBaseType.DECIMAL) {
+      throw new KsqlException("The schema provider method for Abs expects a BigDecimal parameter"
```
Comment (stevenpyzhang) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java]: typo? should say for Ceil

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Floor.java [hunk 4]
```diff
@@ -18,34 +18,51 @@
...
+    return val == null
+        ? null
+        : val.setScale(0, RoundingMode.FLOOR).setScale(val.scale(), RoundingMode.UNNECESSARY);
+  }
+
+  @UdfSchemaProvider
+  public SqlType floorDecimalProvider(final List<SqlType> params) {
+    final SqlType s = params.get(0);
+    if (s.baseType() != SqlBaseType.DECIMAL) {
+      throw new KsqlException("The schema provider method for Abs expects a BigDecimal parameter"
```
Comment (stevenpyzhang) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Floor.java]: copy paste error, should be for `floor`

File: ksql-engine/src/main/java/io/confluent/ksql/function/InternalFunctionRegistry.java [hunk 5]
```diff
@@ -29,7 +29,7 @@
 import io.confluent.ksql.function.udf.UdfMetadata;
 import io.confluent.ksql.function.udf.json.ArrayContainsKudf;
 import io.confluent.ksql.function.udf.json.JsonExtractStringKudf;
-import io.confluent.ksql.function.udf.math.CeilKudf;
+import io.confluent.ksql.function.udf.math.Ceil;
```
Comment (stevenpyzhang) [inline ksql-engine/src/main/java/io/confluent/ksql/function/InternalFunctionRegistry.java]: Is this import used in the file?

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java (hunk 1 above)
Comment (agavra) [reply ksql-engine/src/main/java/io/confluent/ksql/function/udf/math/Ceil.java]: so many derps 🤦‍♂ - this is what I get for thinking this was an "easy PR" and not paying attention to details. Thanks for catching them @stevenpyzhang!

**Pull request merged**